packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -143,28 +143,100 @@ class FileAccessLog extends AccessLog {
    * @default - no file based access logging
    */
   public readonly filePath: string;
+  public readonly virtualNodeLoggingFormat?: CfnVirtualNode.LoggingFormatProperty;
+  public readonly virtualGatewayLoggingFormat?: CfnVirtualGateway.LoggingFormatProperty;
 
-  constructor(filePath: string) {
+  constructor(filePath: string, loggingFormat?: LoggingFormat) {
     super();
     this.filePath = filePath;
+    // For now we have the same setting for Virtual Gateway and Virtual Nodes
+    this.virtualGatewayLoggingFormat = loggingFormat;
+    this.virtualNodeLoggingFormat = loggingFormat;
   }
 
   public bind(_scope: Construct): AccessLogConfig {
     return {
       virtualNodeAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualNodeLoggingFormat,
         },
       },
       virtualGatewayAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualGatewayLoggingFormat,
         },
       },
     };
   }
 }
 
+/**
+ * Configuration for Envoy logging format
+ */
+export class LoggingFormat {
+  /**
+   * Generate logging format from text pattern
+   */
+  public static fromText(text: string): LoggingFormat {
+    return new LoggingFormat(text);
+  }
+  /**
+   * Generate logging format from json key pairs
+   */
+  public static fromJson(jsonPairs: string[][]): LoggingFormat {
+    const json: JsonFormatRef[] = [];
+    if (jsonPairs.length == 0) {
+      throw new Error('Json key pairs cannot be empty.');
+    }
+
+    jsonPairs.forEach(pair => {
+      if (pair.length != 2) {
+        throw new Error('key value pair should be a string array of length 2.');
```

Comment:
If it should be a string array of length 2, why don't we make the type of this thing `{ [key: string]: string }`? It's more natural to express key-value pairs as

```
{
  key: 'value',
  key2: 'value2',
}
```

Than

```
[ ['key', 'value'], ['key2', 'value2'] ]

Reply:
Great suggestion. Nice learn, thanks